From 1ed171530f50a52f6e257d64d04e4a0cf9b27562 Mon Sep 17 00:00:00 2001
From: Joe Maples <joe@frap129.org>
Date: Tue, 24 Oct 2017 21:33:21 +0200
Subject: [PATCH 02/18] hwui: Fix HWUI_COMPILE_FOR_PERF for non-arm

Just fixing an upstream TODO that we can use to our advantage.

TODO: Look into x86 flags to add

Inspired by a change by Alex Nadis (@TheCrazyLex)

Change-Id: Ifb7c8d6498f124980a9099802cb09798fe0492c5
Signed-off-by: Joe Maples <joe@frap129.org>

Conflicts:
	libs/hwui/Android.bp
---
 libs/hwui/Android.bp | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/libs/hwui/Android.bp b/libs/hwui/Android.bp
index 89cb194c3ecd..3cfaeff94f37 100644
--- a/libs/hwui/Android.bp
+++ b/libs/hwui/Android.bp
@@ -82,11 +82,18 @@ cc_defaults {
 
 cc_defaults {
     name: "hwui_compile_for_perf",
-    // TODO: Non-arm?
     cflags: [
         "-fno-omit-frame-pointer",
-        "-marm",
     ],
+
+    arch: {
+        arm: {
+            cflags: ["-marm"],
+        },
+        arm64: {
+            cflags: ["-marm"],
+        },
+    },
 }
 
 cc_defaults {
-- 
2.17.1

